File: app/src/main/java/com/hazizz/droid/Communication/responsePojos/taskPojos/PojoCreator.java
package com.hazizz.droid.communication.responsePojos.taskPojos;

import com.hazizz.droid.communication.responsePojos.Pojo;

import lombok.Data;

@Data
public class PojoCreator implements Pojo {

    private long id;
    private String username;
    private String displayName;
    private String registrationDate;

    public PojoCreator(long id, String username, String displayName, String registrationDate){
        this.id = id;
        this.username = username;
        this.displayName = displayName;
        this.registrationDate = registrationDate;
    }

}

File: lib/communication/pojos/PojoCreator.dart
import 'package:json_annotation/json_annotation.dart';

import 'Pojo.dart';

part 'PojoCreator.g.dart';

@JsonSerializable()
class PojoCreator implements Pojo {

  int id;
  String username;
  String displayName;
  String registrationDate;

  PojoCreator(this.id, this.username, this.displayName, this.registrationDate);

  factory PojoCreator.fromJson(Map<String, dynamic> json) =>
      _$PojoCreatorFromJson(json);

  Map<String, dynamic> toJson() => _$PojoCreatorToJson(this);

}
